kms/cloudkms/decrypter.go

```diff
@@ -0,0 +1,120 @@
+package cloudkms
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"errors"
+	"fmt"
+	"hash/crc32"
+	"io"
+
+	"cloud.google.com/go/kms/apiv1/kmspb"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+
+	"go.step.sm/crypto/kms/apiv1"
+	"go.step.sm/crypto/pemutil"
+)
+
+// CreateDecrypter implements the apiv1.Decrypter interface and returns
+// a crypto.Decrypter backed by a decryption key in Google Cloud KMS.
+func (k *CloudKMS) CreateDecrypter(req *apiv1.CreateDecrypterRequest) (crypto.Decrypter, error) {
+	if req.DecryptionKey == "" {
+		return nil, errors.New("decryption key cannot be empty")
+	}
+	return NewDecrypter(k.client, req.DecryptionKey)
+}
+
+// Decrypter implements a crypto.Decrypter using Google Cloud KMS.
+type Decrypter struct {
+	client        KeyManagementClient
+	decryptionKey string
+	publicKey     crypto.PublicKey
+}
+
+// NewDecrypter creates a new crypto.Decrypter backed by the given
+// Google Cloud KMS decryption key.
+func NewDecrypter(client KeyManagementClient, decryptionKey string) (*Decrypter, error) {
+	// Make sure that the key exists.
+	decrypter := &Decrypter{
+		client:        client,
+		decryptionKey: decryptionKey,
+	}
+	if err := decrypter.preloadKey(decryptionKey); err != nil { // TODO(hs): (option for) lazy load instead?
+		return nil, err
+	}
```

Comment:
Preloading the key makes sure that the key exists, and this is a fast operation.

Reply:
Yes, preloading absolutely makes sense. I'm not saying that we should remove that. Maybe "lazy load" isn't the right word for what I mean. Currently a failure to preload the key, for example due to a connection error, will result in the CA not starting, for example. It might be an option to not make this a hard fail immediately, but to try best effort to preload the key, and if it fails, to try it again when a key operation is required, for example. 

There's value in doing the preload and return an error, because it ensures the key is there, and there were no errors, so the decrypter (and signer) should be usable. But maybe we can decouple that operation, so that the user of the package can decide what semantics they want. For example, a failure to preload the public key could be handled when a call to `Public()` happens, so that the error can be logged instead of blocking CA initialization.